TicketsDemo.Domain/DefaultImplementations/TicketServiceLoggingDecorator.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    abstract class TicketServiceLoggingDecorator : ITicketService
+    {
+        private ITicketService _decoratedobject;
+        private ILogger _logger;
+        public TicketServiceLoggingDecorator(ITicketService service, FileLogger logger)
```

Comment:
reference implementation FileLogger instead of ILogger in the list of dependencies